page_iterator.go

```diff
@@ -46,7 +46,7 @@ func (p *PageResult) getNextLink() *string {
 //
 // It has three parameters. res is the graph response from the initial request and represents the first page.
 // reqAdapter is used for getting the next page and constructorFunc is used for serializing next page's response to the specified type.
-func NewPageIterator(res interface{}, reqAdapter GraphRequestAdapterBase, constructorFunc serialization.ParsableFactory) (*PageIterator, error) {
+func NewPageIterator(res interface{}, reqAdapter abstractions.RequestAdapter, constructorFunc serialization.ParsableFactory) (*PageIterator, error) {
 	page, err := convertToPage(res)
```

Comment:
since this now relies on an interface instead of a struct, nil can be passed and it'd derail things. Would you mind checking for nil and returning an error?